### DISCUSSION TITLE
How to read huge sdf files

### DISCUSSION THREAD

**Discussion by smallrainbow** [epochpic/epoch#196] [Q&A]
Dear developers, 

If the size of the sdf files larger than the available memory size on the postprocessing machine, How to read the variables in the structure part by part?

E.g., use the GetDataSDF.m in epoch-v4.17.15/SDF/Matlab/

If there are some other ways (e.g. python, IDL etc.), it will be fine too.

Thank you!

**Comment by smallrainbow** [comment]
I find a possible way: just read the data block by block (e.g., the total size of the sdf file is 300GB which has nblocks=13, we may add a keyword to the function q=GetDataSDF(filename,**block_num**), where block_num=1:nblocks) : )

An example:

[data,nblocks]=GetDataSDF(input,1,5)
data = 
  struct with fields:

              step: 13064
              time: 0.051202179693252
          Walltime: 6.663639887309074e+04
    Electric_Field: [1×1 struct]
nblocks =
    13

**Reply by liuxybeta** [reply]
Thx!!!
